modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -839,131 +839,114 @@ func (suite *KeeperTestSuite) TestChanUpgradeConfirm() {
 	}
 }
 
-// TODO: Uncomment and address testcases when appropriate, timeout logic currently causes failures
-// func (suite *KeeperTestSuite) TestChanUpgradeOpen() {
-// 	var path *ibctesting.Path
-// 	testCases := []struct {
-// 		name     string
-// 		malleate func()
-// 		expError error
-// 	}{
-// 		{
-// 			"success",
-// 			func() {},
-// 			nil,
-// 		},
-// 		{
-// 			"channel not found",
-// 			func() {
-// 				path.EndpointA.ChannelConfig.PortID = ibctesting.InvalidID
-// 			},
-// 			types.ErrChannelNotFound,
-// 		},
+func (suite *KeeperTestSuite) TestChanUpgradeOpen() {
+	var path *ibctesting.Path
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			"success",
+			func() {},
+			nil,
+		},
+		{
+			"channel not found",
+			func() {
+				path.EndpointA.ChannelConfig.PortID = ibctesting.InvalidID
+			},
+			types.ErrChannelNotFound,
+		},
+		{
+			"channel state is not FLUSHCOMPLETE",
+			func() {
+				channel := path.EndpointA.GetChannel()
+				channel.State = types.STATE_FLUSHING
+				path.EndpointA.SetChannel(channel)
+			},
+			types.ErrInvalidChannelState,
+		},
+		{
+			"connection not found",
+			func() {
+				channel := path.EndpointA.GetChannel()
+				channel.ConnectionHops = []string{"connection-100"}
+				path.EndpointA.SetChannel(channel)
+			},
+			connectiontypes.ErrConnectionNotFound,
+		},
+		{
+			"invalid connection state",
+			func() {
+				connectionEnd := path.EndpointA.GetConnection()
+				connectionEnd.State = connectiontypes.UNINITIALIZED
+				path.EndpointA.SetConnection(connectionEnd)
+			},
+			connectiontypes.ErrInvalidConnectionState,
+		},
+		{
+			"invalid counterparty channel state",
+			func() {
+				channel := path.EndpointB.GetChannel()
+				channel.State = types.CLOSED
+				path.EndpointB.SetChannel(channel)
+			},
+			types.ErrInvalidCounterparty,
```

Comment:
this is a new test to ensure the error propagates in the case of an invalid counterparty channel state